test/commands/auth/jwt/grant.test.ts

```diff
@@ -0,0 +1,342 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { $$, expect, test } from '@salesforce/command/lib/test';
+import {
+  Aliases,
+  AuthFields,
+  AuthInfo,
+  Config,
+  SfdxError
+} from '@salesforce/core';
+import { MockTestOrgData } from '@salesforce/core/lib/testSetup';
+
+interface Options {
+  authInfoCreateFails?: boolean;
+  existingAuth?: boolean;
+}
+
+describe('auth:jwt:grant', async () => {
+  const testData = new MockTestOrgData();
+  const spies = new Map();
+  let authFields: AuthFields;
+
+  afterEach(() => spies.clear());
+
+  async function prepareStubs(options: Options = {}) {
+    authFields = await testData.getConfig();
+    const baseAuthConfig = {
+      getFields: () => authFields,
+      getConnectionOptions: () => ({ accessToken: authFields.accessToken }),
+      isJwt: () => true,
+      isOauth: () => false,
+      save: async () => { },
+      update: () => {
+        return { save: async () => { } };
+      }
+    };
+
+    if (options.authInfoCreateFails) {
+      $$.SANDBOX.stub(AuthInfo, 'create').throws(
+        new Error('invalid client id')
+      );
+    } else if (options.existingAuth) {
+      $$.SANDBOX.stub(AuthInfo, 'create')
+        .onFirstCall()
+        .throws(new SfdxError('auth exists', 'AuthInfoOverwriteError'))
+        .onSecondCall()
+        // @ts-ignore
```

Comment:
You can get rid of these by using `stubInterface` from `@salesfroce/ts-sinon`. For example, `const baseAuthConfig = stubInterface<AuthInfo>({...});`

Or, another way to get rid of it is using `stubMethod($$.SANDBOX, AuthInfo, 'create')` also from ts-sinon. I like the first solution because it actually mirrors the type, but this is still better than adding ts-ignore.